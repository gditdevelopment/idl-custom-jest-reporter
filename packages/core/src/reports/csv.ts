import { TestResult } from '../core/jest-mapping';
const { createObjectCsvWriter } = require('csv-writer');

export const saveCsvReport = async (testResults: TestResult[], path: string) => {
  const csvWriter = createObjectCsvWriter({
    path,
    header: [
      { id: 'testName', title: 'TESTNAME' },
      { id: 'suiteName', title: 'SUITENAME' },
      { id: 'duration', title: 'DURATION' },
      { id: 'status', title: 'STATUS' },
    ],
  });

  return csvWriter.writeRecords(testResults);
};
