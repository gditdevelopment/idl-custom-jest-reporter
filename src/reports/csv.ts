import { TestResult } from '../core/jest-mapping';
const { createObjectCsvWriter } = require('csv-writer');

export const saveCsvReport = async (testResults: TestResult[], path: string) => {
  const csvWriter = createObjectCsvWriter({
    path,
    alwaysQuote: true,
    header: [
      { id: 'all', title: 'ALL' },
      { id: 'testName', title: 'TESTNAME' },
      { id: 'suiteName', title: 'SUITENAME' },
      { id: 'duration', title: 'DURATION' },
      { id: 'status', title: 'STATUS' },
      { id: 'numSuitePassingTests', title: 'SUITEPASSINGTESTS' },
      { id: 'numSuiteFailingTests', title: 'SUITEFAILINGTESTS' },
      { id: 'numSuiteSkippedTests', title: 'SUITESKIPPEDTESTS' },
      { id: 'suiteStartTime', title: 'SUITESTARTTIME' },
      { id: 'suiteEndTime', title: 'SUITEENDTIME' },
      { id: 'suiteDuration', title: 'SUITEDURATION' },
      { id: 'suiteSlowWarning', title: 'SUITESLOWWARNING' },
      { id: 'failureMessage', title: 'FAILUREMESSAGE' },
      { id: 'testFilePath', title: 'TESTFILEPATH' },
      { id: 'testLocationInResults', title: 'TESTLOCATIONINRESULTS' }
    ]
  });

  return csvWriter.writeRecords(testResults);
};
