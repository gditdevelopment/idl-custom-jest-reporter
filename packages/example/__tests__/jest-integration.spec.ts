import { createReadStream, readFileSync } from 'fs';
import csv from 'csv-parser';
import { doesNotMatch } from 'assert';

const roundOffNearest100 = (num) => {
  return Math.floor(num / 100) * 100;
};

test('that the json report works', () => {
  const report = JSON.parse(readFileSync('performance-report.json').toString());

  expect(
    report.map(({ fullName, status, duration }) => ({
      fullName,
      status,
      duration: roundOffNearest100(duration),
    }))
  ).toMatchSnapshot();
});

test('that the csv report works', (done) => {
  const rows = [];
  createReadStream('performance-report.csv')
    .pipe(csv())
    .on('data', (row) => {
      rows.push({ ...row, DURATION: roundOffNearest100(row.DURATION) });
    })
    .on('end', () => {
      expect(rows).toMatchSnapshot();
      done();
    });
});
