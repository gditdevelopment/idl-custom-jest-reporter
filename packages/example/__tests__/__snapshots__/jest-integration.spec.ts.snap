// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`that the csv report works 1`] = `
Array [
  Object {
    "DURATION": 0,
    "NAME": "Fruits should have a raspberry",
    "STATUS": "passed",
  },
  Object {
    "DURATION": 500,
    "NAME": "Fruits should have an apple",
    "STATUS": "passed",
  },
  Object {
    "DURATION": 0,
    "NAME": "should have a raspberry without describe",
    "STATUS": "passed",
  },
  Object {
    "DURATION": 0,
    "NAME": "Calculator should sum 2+3",
    "STATUS": "passed",
  },
  Object {
    "DURATION": 3000,
    "NAME": "Calculator should sum 8+0, with timeout",
    "STATUS": "passed",
  },
]
`;

exports[`that the json report works 1`] = `
Array [
  Object {
    "duration": 0,
    "fullName": "Fruits should have a raspberry",
    "status": "passed",
  },
  Object {
    "duration": 500,
    "fullName": "Fruits should have an apple",
    "status": "passed",
  },
  Object {
    "duration": 0,
    "fullName": "should have a raspberry without describe",
    "status": "passed",
  },
  Object {
    "duration": 0,
    "fullName": "Calculator should sum 2+3",
    "status": "passed",
  },
  Object {
    "duration": 3000,
    "fullName": "Calculator should sum 8+0, with timeout",
    "status": "passed",
  },
]
`;
